tests/sentry/sentry_metrics/test_batch.py

```diff
@@ -254,6 +325,135 @@ def test_all_resolved(caplog, settings):
+                "session.status": 5,
+            }
+        },
+        {
+            1: {
+                "c:sessions/session@none": Metadata(id=1, fetch_type=FetchType.CACHE_HIT),
+                "d:sessions/duration@second": Metadata(id=2, fetch_type=FetchType.CACHE_HIT),
+                "environment": Metadata(id=3, fetch_type=FetchType.CACHE_HIT),
+                "s:sessions/error@none": Metadata(id=4,
+                                                  fetch_type=FetchType.CACHE_HIT),
+                "session.status": Metadata(id=5,
+                                           fetch_type=FetchType.CACHE_HIT),
+            }
+        },
+    )
+
+    assert _get_string_indexer_log_records(caplog) == []
+    assert _deconstruct_messages(snuba_payloads) == [
+        (
+            {
+                "mapping_meta": {
+                    "c": {
+                        "1": "c:sessions/session@none",
+                        "3": "environment",
+                        "5": "session.status",
+                    },
+                },
+                "metric_id": 1,
+                "org_id": 1,
+                "project_id": 3,
+                "retention_days": 90,
+                "tags": {"3": "production", "5": "init"},
+                "timestamp": ts,
+                "type": "c",
+                "use_case_id": "performance",
+                "value": 1.0,
+            },
+            [("mapping_sources", b"c"), ("metric_type", "c")],
+        ),
+        (
+            {
+                "mapping_meta": {
+                    "c": {
+                        "2": "d:sessions/duration@second",
+                        "3": "environment",
+                        "5": "session.status",
+                    },
+                },
+                "metric_id": 2,
+                "org_id": 1,
+                "project_id": 3,
+                "retention_days": 90,
+                "tags": {"3": "production", "5": "healthy"},
+                "timestamp": ts,
+                "type": "d",
+                "unit": "seconds",
+                "use_case_id": "performance",
+                "value": [4, 5, 6],
+            },
+            [("mapping_sources", b"c"), ("metric_type", "d")],
+        ),
+        (
+            {
+                "mapping_meta": {
+                    "c": {
+                        "3": "environment",
+                        "4": "s:sessions/error@none",
+                        "5": "session.status",
+                    },
+                },
+                "metric_id": 8,
+                "org_id": 1,
+                "project_id": 3,
+                "retention_days": 90,
+                "tags": {"3": "production", "5": "errored"},
+                "timestamp": ts,
+                "type": "s",
+                "use_case_id": "performance",
+                "value": [3],
+            },
+            [("mapping_sources", b"c"), ("metric_type", "s")],
+        ),
+    ]
+
+
+@pytest.mark.django_db
```

Comment:
These marks are required since the code now uses options which depends on the database